src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -1,10 +1,23 @@
 import {Logger, SfdxError} from '@salesforce/core';
-import {Catalog, Rule, RuleGroup, RuleResult, RuleTarget, RuleViolation, ESRule, ESReport, ESMessage} from '../../types';
+import {Catalog, LooseObject, Rule, RuleGroup, RuleResult, RuleTarget, RuleViolation, ESRule, ESReport, ESMessage} from '../../types';
 import {OutputProcessor} from '../pmd/OutputProcessor';
 import {RuleEngine} from '../services/RuleEngine';
 import {CLIEngine} from 'eslint';
 import * as path from 'path';
 
+// TODO: We recognize that this fix for W-7791882, including DEFAULT_ENV_VARS and the overrideDefaultEnv() method, is GARBAGE.
+//       During the 3.0 release cycle, we must replace it a better handling of the eslintrc file in general.
+// These are the environment variables that we'll want enabled by default in our ESLint baseConfig.
+let DEFAULT_ENV_VARS: LooseObject = {
```

Comment:
Could this use the existing engineOptions pattern that configures the engine per invocation of the #run method? 

The code is currently using a static variable which could have unintended consequences and breaks the class encapsulation.

Reply:
Oh, that's a good point and I totally missed it.
+1 to using existing engineOptions maps to flow in commandline flag values.

Reply:
Good point. I'll see what I can do.